test/e2e/dnscheck_test.go

```diff
@@ -65,53 +63,7 @@ var _ = Describe("DNS checker metrics", Ordered, ContinueOnFailure, func() {
 			}
 			GinkgoWriter.Printf("Found healthy DNS checker metric for %v\n", foundCheckers)
 			return true
-		}, "30s", "5s").Should(BeTrue(), "DNS checker metrics did not report healthy status within the timeout period")
-	})
-
-	It("should report unhealthy status for CoreDNS checkers when CoreDNS pods are not ready", func() {
-		By("Getting the CoreDNS deployment")
-		deployment, err := getCoreDNSDeployment(clientset)
```

Comment:
Removed this check because it is not compatible with AKS. A controller will automatically override the scaled down replica count and set it back to the original amount. This pretty much guarantees there will always be some core dns pods ready and the checker will not encounter this error.